### PR TITLE
[JENKINS-56499] Remove synchronized on method updateUsageForId for API token usage stat

### PR DESCRIPTION
**Pull request by liucc52**
See [JENKINS-56499](https://issues.jenkins-ci.org/browse/JENKINS-56499).

### Proposed changelog entries

* Entry 1: Issue, There is a synchronized method on API token usage stat, which will cause thread hang on
* ...

### Submitter checklist

- [ ] JIRA issue is well described
- [ ] Changelog entry appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
      * Use the `Internal: ` prefix if the change has no user-visible impact (API, test frameworks, etc.)
- [ ] Appropriate autotests or explanation to why this change has no tests
- [ ] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@mention

### CODE REVIEW

File: core/src/main/java/jenkins/security/apitoken/ApiTokenPropertyConfiguration.java [hunk 1]
```diff
@@ -53,7 +53,7 @@
      * Each time an API Token is used, its usage counter is incremented and the last used date is updated.
      * You can disable this feature using this property.
      */
-    private boolean usageStatisticsEnabled = true;
+    private boolean usageStatisticsEnabled = false;
```
Comment (daniel-beck) [inline core/src/main/java/jenkins/security/apitoken/ApiTokenPropertyConfiguration.java]: This is lacking justification.

**Comment by daniel-beck** [comment]
It is unclear to me why this would change anything. The Jira issue isn't described well enough to show.

**Comment by liucc52** [comment]
> is unclear to me why this would change anything. The Jira issue is
This will remove the synchronize if The usage statistics is disabled .

**Review by oleg-nenashev — APPROVED** [review]
LGTM. It makes sense to do the same with the `findTokenStatsById()` call

**Comment by daniel-beck** [comment]
@liucc52 

>  This remove the synchronize if The usage statistics is disabled .

I don't see how the Jira issue shows that this change would fix the problem.

**Comment by oleg-nenashev** [comment]
@daniel-beck I am not sure whether the patch fixes JENKINS-56499 entirely. But reducing the lock scope is definitely something which addresses "Actually we do not need to record api call stat in **SOME** situation." in the bottom of the ticket. So I would suggest accepting the pull request

**Comment by oleg-nenashev** [comment]
@daniel-beck in IRC: "no opinion, merge if you like it"

**Review by alecharp — APPROVED** [review]

**Comment by oleg-nenashev** [comment]
Will merge it tomorrow if no negative feedback

**Review by fcojfernandez — APPROVED** [review]

**Review by MRamonLeon — APPROVED** [review]

**Comment by oleg-nenashev** [comment]
Thanks @liucc52 !

**Pull request merged**